language/move-prover/src/spec_translator.rs

```diff
@@ -1475,8 +1462,14 @@ impl<'env> SpecTranslator<'env> {
     fn translate_resource_access(&self, node_id: NodeId, args: &[Exp]) {
         self.trace_value(node_id, TraceItem::Exp, || {
             let rty = &self.module_env().get_node_instantiation(node_id)[0];
-            let type_value = self.translate_type(rty);
-            emit!(self.writer, "$ResourceValue($m, {}, ", type_value);
+            let (mid, sid, targs) = rty.require_struct();
```

Comment:
Just trying to wrap my head around what is going on.  It looks like spec functions can access global memory.  Since the memory is split now, the parts that they access has to be passed in as explicit parameters.  Therefore, the lookup translation will also change.  Is my understanding correct?

Reply:
Yes, all individual memories which are used by the helper function as well as the functions called by it need to be passed as individual parameters. Fortunately, a similar mechanism already existed for spec (ghost) variables, so that one just needed to be extended.